src/terms/magnetic.jl

```diff
@@ -27,7 +27,7 @@ function TermMagnetic(basis::PlaneWaveBasis{T}, Afunction::Function) where {T}
             end
         end
     end
-    TermMagnetic(Apotential)
+    TermMagnetic(to_device.(Ref(basis.architecture), Apotential))
```

Comment:
Did you check that the Magnetic term now works on the GPU. If yes, we maybe can remove the respective comment in docs/src/developer/gpu_computations.md

Reply:
Ah, thanks for your efforts, but I was mostly wondering if they now happen to work by accident ...

Reply:
BTW our main paradigm in DFTK is to keep the code base manageable. That means we really want to avoid duplicating code to the largest extend possible ... and with Julia this very often works. So if you find yourself copying large chunks of code, better first ask for some advice before going through the hassle ;)